nautobot/core/management/commands/pre_migrate.py

```diff
@@ -120,6 +120,48 @@ def check_configcontext_uniqueness():
         )
 
 
+def check_interface_ipaddress_vrf_uniqueness(command):
+    """
+    Check for uniqueness enforcement changes for Interface and VMInterface IPAddress assignments.
+
+    - All IP Addresses assigned to a specific Interface or VMInterface must share the same VRF or have no VRF.
+    """
+
+    failed = False
+
+    command.stdout.write(command.style.WARNING(">>> Running interface/vminterface vrf uniqueness checks..."))
+
+    for ip_address_assignment in (
+        IPAddress.objects.values("assigned_object_type", "assigned_object_id")
+        .annotate(
+            vrf_count=models.Count("vrf_id", distinct=True),
+            vrf_null_count=models.Count("pk", filter=models.Q(vrf_id__isnull=True)),
+        )
+        .order_by()
```

Comment:
What does `.order_by()` do here?

Reply:
Gets rid of any custom sorting defined on the model so that the annotation doesn't get messed up by it